File: C/AD-HOC/1136 Bingo!.c
#include <stdio.h>
#include <math.h>

int checkExistence(int num, int len, int *numFromCombs)
{
    for (int i = 0; i < len; i++)
        if (numFromCombs[i] == num)
        {
            return 1;
        }
    return 0;
}

int draw(int comb, int numBalls)
{
    int balls[numBalls];
    for (int i = 0; i < numBalls; i++)
        scanf("%d", &balls[i]);

    int len = 1, num, numFromCombs[comb];
    numFromCombs[0] = 0;
    for (int i = 0; i < numBalls; i++)
    {
        for (int j = 0; j < numBalls; j++)
        {
            num = abs(balls[i] - balls[j]);
            if (num <= comb)
            {
                if (checkExistence(num, len, numFromCombs) == 0) // WITHOUT REPETITIONS IN THE ARRAY
                {
                    numFromCombs[len] = num;
                    len += 1;
                }
            }
            if (len == (comb + 1)) // SAVE TIME
            {
                printf("Y\n");
                return 0;
            }
        }
        if (len == (comb + 1)) // SAVE TIME
        {
            printf("Y\n");
            return 0;
        }
    }
    if (len == (comb + 1)) // SAVE TIME
    {
        printf("Y\n");
        return 0;
    }
    else
    {
        printf("N\n");
        return 0;
    }
}

int main()
{
    int combination, numBalls;
    scanf("%d %d", &combination, &numBalls);
    while ((combination != 0) || (numBalls != 0))
    {
        draw(combination, numBalls);
        scanf("%d %d", &combination, &numBalls);
    }

    return 0;
}

File: Python/AD-HOC/1136 Bingo!.py
# SUBPROGRAM
def partition(balls, left, right):
    pivot = balls[right]
    i = left - 1
    for j in range(left, right):
        if balls[j] <= pivot:
            i = i+1
            balls[i], balls[j] = balls[j], balls[i]
    balls[i+1], balls[right] = balls[right], balls[i+1]
    return i+1

def quicksort(balls, left, right):
    if left < right:
        pivot = partition(balls, left, right)
        quicksort(balls, left, pivot-1)
        quicksort(balls, pivot+1, right)
    return balls

def draw(comb, numBalls):
    balls = list(map(int, input().split()))
    balls = quicksort(balls, 0, (len(balls)-1))
    numCombs = list()
    numCombs += [0]
    while len(balls) != 0:
        for j in range(0, len(balls)):
            num = abs(balls[0] - balls[j])
            if num <= comb:
                if num not in numCombs:
                    numCombs += [num]
            if len(numCombs) == (comb+1):
                print("Y")
                return None
        if len(numCombs) == (comb+1):
            print("Y")
            return None
        balls.pop(0)
    if len(numCombs) == (comb+1):
        print("Y")
        return None
    else:
        print("N")
        return None

# MAIN
combination, numBalls = map(int, input().split())
while combination != 0 or numBalls != 0:
    draw(combination, numBalls)
    combination, numBalls = map(int, input().split())